### PR TITLE
Use eslint-plugin-react@7.18.3

### PR DESCRIPTION
**Pull request by whymarrh**
This PR updates our `eslint-plugin-react` version to address a warning on install:

```
warning " > eslint-plugin-react@7.5.1" has incorrect peer dependency "eslint@^3.0.0 || ^4.0.0".
```

It also looks like this plugin has gotten better at detecting the weird ways we alias props, e.g.:

```js
const state = this.props
const identity = state.identity || {}
const account = state.account || {}
const isFauceting = state.isFauceting
```

### CODE REVIEW

File: ui/app/pages/first-time-flow/create-password/new-account/new-account.component.js [hunk 1]
```diff
@@ -211,7 +211,7 @@ export default class NewAccount extends PureComponent {
                 rel="noopener noreferrer"
               >
                 <span className="first-time-flow__link-text">
-                  { 'Terms of Use' }
+                  Terms of Use
```
Comment (rickycodes) [inline ui/app/pages/first-time-flow/create-password/new-account/new-account.component.js]: should we be using translations for these?

Comment (whymarrh) [reply ui/app/pages/first-time-flow/create-password/new-account/new-account.component.js]: We should, yes

Comment (whymarrh) [reply ui/app/pages/first-time-flow/create-password/new-account/new-account.component.js]: Fixed

**Review by Gudahtt — APPROVED** [review]
LGTM!

**Review by rekmarks — APPROVED** [review]

**Review by rickycodes — APPROVED** [review]

**Pull request merged**